jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -205,14 +206,9 @@ public void onComplete(Result result)
 
                 conversation.setAttribute(authenticationAttribute, true);
 
-                URI requestURI = request.getURI();
-                String path = null;
-                if (requestURI == null)
-                {
-                    requestURI = resolveURI(request, null);
-                    path = request.getPath();
-                }
-                Request newRequest = client.copyRequest(request, requestURI);
+                Request newRequest = client.copyRequest(request, request.getURI());
+                if (HttpMethod.CONNECT.is(newRequest.getMethod()))
+                    newRequest.path(request.getPath());
```

Comment:
I have a slight preference for something like
```suggestion
                Request newRequest = (HttpMethod.CONNECT.is(request.getMethod()))
                    ? client.copyRequest(request, request.getPath());
                    : client.copyRequest(request, request.getURI());
```

Or if the right copyRequest semantic can't be easily provided, then at least below would be a little clearer to me:

```suggestion
                Request newRequest = client.copyRequest(request, request.getURI());
                if (HttpMethod.CONNECT.is(request.getMethod()))
                    newRequest.path(request.getPath());
```

Reply:
The first proposed change is not correct.

The second proposed change just replaces the variable `newRequest` with `request`, which I don't see how it is clearer?

Reply:
@sbordet the first proposal is pseudo code: check the method of the original request first and then build the new request appropriately.

The second proposal is clearer to me, as I don't have to do the double take of thinking: oh it is conditional on its own method... oh what is its method, oh it copied it from the original request, ah so it is just the method of the original request.   It is clearer just to make it conditional on the method of the original request.